.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,120 @@
+name: Test and Publish NuGet
+
+on:
+  push:
+    branches:
+      - master
+      - development
+      - simple
+      - simple-test2
+    paths-ignore:
+      - '.github/workflows/docs.yml'
+      - 'docs/**'
+  pull_request:
+    branches:
+      - master
+      - development
+    paths-ignore:
+      - '.github/workflows/docs.yml'
+      - 'docs/*'
+  workflow_dispatch: 
+
+concurrency:
+  group: ${{github.workflow}}-${{github.event.pull_request.number || github.ref}}
+  cancel-in-progress: true
+
+env:
+  # Right now we can't run experimental tests that can fail, since it will affect the PR
+  # By default we only run it on master/development branches
+  # Unless the variable will be true
+  EXPERIMENTAL: false
+
+  # You are interested in changing these values
+  BUILD_CONFIGURATIONS:                 "['Release']"
+  DOTNET_TARGET_FRAMEWORKS:             "['netcoreapp3.1', 'net5.0', 'net6.0', 'net7.0', 'net8.0']"
+  DOTNET_TARGET_FRAMEWORKS_MACOS_ARM64: "['net6.0', 'net7.0', 'net8.0']"
+  FRAMEWORK_TARGET_FRAMEWORKS:          "['net35', 'net48']"
+
+jobs:
+
+  # This technique allows us to run `test-results-publish.yml` with the owner's secrets
+  # without comprimizing the security
+  upload-event-file:
+    name: Upload Event File
+    runs-on: ubuntu-latest
+    steps:
+      - name: Upload event file
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-event-file
+          path: ${{ github.event_path }}
+          retention-days: 1
```

Comment:
Unsure about this, should these be exposed to the public?

Reply:
The event file doesn't have any private information, it's just a webhook payload.